service/src/main/resources/api/service_openapi.yaml

```diff
@@ -312,7 +312,26 @@ paths:
           $ref: '#/components/responses/NotFound'
         '500':
           $ref: '#/components/responses/ServerError'
-  ## END OF DEPRECATED PATHS ##
+
+  /api/workspaces/v1/{workspaceId}/resources/referenced/{resourceId}:
+    parameters:
+    - $ref: '#/components/parameters/WorkspaceId'
+    - $ref: '#/components/parameters/ResourceId'
+    get:
+      summary: Validate the existence of a referenced resource
+      operationId: validateReference
+      tags: [Resource]
+      responses:
+        '200':
+          $ref: '#/components/responses/ValidateReferenceResponse'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+        '404':
+          $ref: '#/components/responses/NotFound'
```

Comment:
Can you point me where these 403 and 404 errors are thrown? In the `ResourceController` class, it looks like it's returning a boolean if it exists or not. I'm not clear on when to expect a `false` vs one of these codes.

Reply:
Yep, this comes from the call to `validateWorkspaceAndAction` ([link](https://github.com/DataBiosphere/terra-workspace-manager/blob/zl-pf702/service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java#L115)) inside `checkAccess`, which is common set of checks before a lot of WSM endpoints.

That will throw a 404 if the workspace doesn't exist, or a 403 if it does exist and the user doesn't have the right permission (in this case, at least read access). If the workspace does exist and the user does have read access, the response should be a `true/false` (or any unexpected error from the cloud/TDR)

Reply:
Ah okay, so the error codes apply to the workspace and the response code applies to the resource. Makes sense.